src/client.cpp

```diff
@@ -773,11 +765,8 @@ void CClient::OnHandledSignal ( int sigNum )
     {
     case SIGINT:
     case SIGTERM:
-        // if connected, terminate connection (needed for headless mode)
-        if ( IsRunning() )
-        {
-            Stop();
-        }
+        // if connected, Stop client (needed for headless mode)
+        Stop();
```

Comment:
Client::Stop is a slot which looks like it's meant for handling the CChannel::Disconnected signal.

We shouldn't conflate concepts like this.  Having the (command line interface) send a "stop" signal isn't the same as the network saying it's lost the connection.  `QCoreApplication::instance()->exit()` _should_ be enough here, with appropriate `OnAboutToQuit` handlers.

Reply:
So
- Sound.OnAboutToQuit would call Sound.Stop.
- Channel.OnAboutToQuit would call Channel.SetEnable ( false )
- Client.OnAboutToQuit would call Client.Stop -- which wouldn't call Sound.Stop or Channel.SetEnable ( false )

Reply:
https://github.com/jamulussoftware/jamulus/issues/3402

Reply:
> Certainly possible, but I'd like to hear @softins opinion here too.

Yes, sorry. This PR is on my list of things to look at again, but just hadn't got to the top of the pile yet! I would certainly like to understand and comment.